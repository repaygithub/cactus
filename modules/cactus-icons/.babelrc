{
  "presets": ["@repay/babel-preset"],
  "sourceMaps": "inline",
  "plugins": [["@babel/plugin-proposal-private-property-in-object", { "loose": true }]]
}