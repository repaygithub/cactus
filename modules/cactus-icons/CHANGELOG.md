# v3.0.1 (Wed Jan 26 2022)

#### 🔩 Dependency Updates

- Cactus 715:: Dependency updates [#686](https://github.com/repaygithub/cactus/pull/686) ([@daniloPenaR](https://github.com/daniloPenaR) [@wilysword](https://github.com/wilysword) [@Dhalton](https://github.com/Dhalton))

#### Authors: 3

- Danilo Peña ([@daniloPenaR](https://github.com/daniloPenaR))
- Dhalton Huber ([@Dhalton](https://github.com/Dhalton))
- Glen Phelps ([@wilysword](https://github.com/wilysword))

---

# v3.0.0 (Mon Nov 15 2021)

#### 💥 Breaking Change

- Default export is now an object containing all icons. IconProps & IconSizes types now exported. [#669](https://github.com/repaygithub/cactus/pull/669) ([@Dhalton](https://github.com/Dhalton))

#### Authors: 1

- Dhalton Huber ([@Dhalton](https://github.com/Dhalton))

---

# v2.6.0 (Mon Jul 19 2021)

#### 🚀 Enhancement

- Allow iconSize prop to accept any valid font size in addition to the tiny/small/medium/etc. values from the theme [#622](https://github.com/repaygithub/cactus/pull/622) ([@mikesoltow](https://github.com/mikesoltow))
- Add DescriptiveDelivered icon [#618](https://github.com/repaygithub/cactus/pull/618) ([@wilysword](https://github.com/wilysword))

#### 🔩 Dependency Updates

- Upgrade various dependencies [#615](https://github.com/repaygithub/cactus/pull/615) ([@mikesoltow](https://github.com/mikesoltow))

#### Authors: 2

- [@mikesoltow](https://github.com/mikesoltow)
- Glen Phelps ([@wilysword](https://github.com/wilysword))

---

# v2.5.0 (Wed Jul 07 2021)

#### 🚀 Enhancement

- Add StatusStarFilled icon [#607](https://github.com/repaygithub/cactus/pull/607) ([@Dhalton](https://github.com/Dhalton))

#### Authors: 1

- Dhalton Huber ([@Dhalton](https://github.com/Dhalton))

---

# v2.4.1 (Thu Mar 04 2021)

#### 🏠 Internal

- Add some test coverage [#532](https://github.com/repaygithub/cactus/pull/532) ([@Dhalton](https://github.com/Dhalton))

#### Authors: 1

- Dhalton Huber ([@Dhalton](https://github.com/Dhalton))

---

# v2.4.0 (Thu Feb 25 2021)

#### 🚀 Enhancement

- Add DescriptivePalette icon [#534](https://github.com/repaygithub/cactus/pull/534) ([@daniloPenaR](https://github.com/daniloPenaR))

#### 🔩 Dependency Updates

- Upgrade Jest SonarQube reporter [#524](https://github.com/repaygithub/cactus/pull/524) ([@mikesoltow](https://github.com/mikesoltow))
- Upgrade various dev dependencies, including build/bundling library [#518](https://github.com/repaygithub/cactus/pull/518) ([@mikesoltow](https://github.com/mikesoltow))

#### Authors: 2

- [@daniloPenaR](https://github.com/daniloPenaR)
- Michael Soltow ([@mikesoltow](https://github.com/mikesoltow))

---

# v2.3.0 (Thu Feb 04 2021)

#### 🚀 Enhancement

- Add NavigationFirst and NavigationLast icons [#504](https://github.com/repaygithub/cactus/pull/504) ([@wilysword](https://github.com/wilysword))

#### Authors: 1

- Glen Phelps ([@wilysword](https://github.com/wilysword))

---

# v2.2.0 (Mon Feb 01 2021)

#### 🚀 Enhancement

- Add ActionsCopy icon [#497](https://github.com/repaygithub/cactus/pull/497) ([@wilysword](https://github.com/wilysword))

#### Authors: 1

- [@wilysword](https://github.com/wilysword)

---

# v2.1.6-beta.0 (Fri Jan 08 2021)

#### 🔩 Dependency Updates

- Upgrade `@repay/cactus-theme` to `v2.0.2-beta-0`

#### Authors: 1

- Mike Soltow ([@mikesoltow](https://github.com/mikesoltow))

---

# v2.1.5 (Mon Dec 21 2020)

#### 🏠 Internal

- Minor refactoring to address SonarQube issues [#454](https://github.com/repaygithub/cactus/pull/454) ([@mikesoltow](https://github.com/mikesoltow))

#### Authors: 1

- Michael Soltow ([@mikesoltow](https://github.com/mikesoltow))

---

# v2.1.3 (Mon Nov 30 2020)

#### 🔩 Dependency Updates

- Upgrade `@repay/cactus-theme` to `v2.0.0`

#### Authors: 1

- Mike Soltow ([@mikesoltow](https://github.com/mikesoltow))

---

# v2.1.3-beta.0 (Thu Nov 19 2020)

#### 🔩 Dependency Updates

- Upgrade `@repay/cactus-theme` to `v2.0.0-beta-0`

#### Authors: 1

- Mike Soltow ([@mikesoltow](https://github.com/mikesoltow))

---

# v2.1.2 (Wed Nov 11 2020)

#### 🐛 Bug Fix

- Add missing `dist` files

#### Authors: 1

- Mike Soltow ([@mikesoltow](https://github.com/mikesoltow))

---

# v2.1.1 (Tue Nov 10 2020)

#### 🔩 Dependency Updates

- Upgrade `@repay/cactus-theme` to `v1.1.0`

#### Authors: 1

- Mike Soltow ([@mikesoltow](https://github.com/mikesoltow))
