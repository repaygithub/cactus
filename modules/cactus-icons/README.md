# `@repay/cactus-icons`

Cactus design-system icons as React elements for web.

## Usage

For usage, see the documentation for the [Cactus Design System Icons](../../docs/Icons/README.md).

## Contributing

Building the repository

```
yarn build
```

Testing the build

```
yarn test
```

You must build before the tests will pass.

### Publishing

In order to provide easily accessible imports and maintain a clean repository, this module is actually published from the completed `dist` folder instead of this root. During the build process, the svgs are converted to TypeScript files in the `built` folder, and then everything is run through babel into the `dist` folder.

This will also affect linking because you will need to link this component from inside there or copy the contents into the root module folder.

The list of currently available icons will be automatically edited on build in the `docs` folder at the repository root.
