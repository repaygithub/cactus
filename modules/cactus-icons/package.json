{
  "name": "@repay/cactus-icons",
  "version": "2.4.1",
  "description": "Cactus design-system icons as React elements",
  "module": "i/index.js",
  "main": "i/index.cjs.js",
  "types": "i/index.d.ts",
  "repository": "https://github.com/repaygithub/cactus/tree/master/modules/cactus-icons",
  "author": "REPAY",
  "license": "MIT",
  "files": [
    "i"
  ],
  "sideEffects": false,
  "scripts": {
    "cleanup": "rm -rf ts i",
    "start": "start-storybook -p 9001 -c .storybook",
    "dev": "echo -e \"This now builds in watch mode, use yarn icons start for storybook\\n\"; repay-scripts dev --lib --tree-shaking --config rollup.config.js ts/index.ts",
    "prebuild": "yarn build:ts",
    "build": "repay-scripts build --lib --tree-shaking --config rollup.config.js ts/index.ts",
    "postbuild": "yarn build:types && node scripts/gen-docs.js",
    "build:ci": "yarn build:ts && yarn build && yarn build:types",
    "build:ts": "node ./scripts/svgr",
    "postbuild:ts": "node ./scripts/post-build-ts",
    "build:types": "tsc -p tsconfig.json --emitDeclarationOnly",
    "build:stories": "build-storybook -c .storybook -o ./.storybook/dist",
    "debug": "node --inspect-brk ./node_modules/.bin/svgr svgs -d built --config-file svgr.config.js --ext tsx",
    "test": "yarn jest",
    "test:local": " yarn test:types && yarn test",
    "test:ci": "yarn test:local --maxWorkers=2 && yarn test:stories",
    "test:types": "tsc -p tsconfig.test.json --noEmit",
    "test:stories": "start-storybook -p 9133 -c .storybook --smoke-test"
  },
  "jest": {
    "transform": {
      "^.+\\.(t|j)sx?$": "babel-jest"
    },
    "moduleNameMapper": {
      "^styled-components": "<rootDir>/../../node_modules/styled-components/dist/styled-components.browser.cjs.js"
    },
    "setupFilesAfterEnv": [
      "<rootDir>/tests/__setup__/setup.ts"
    ],
    "testRegex": ".*\\.test\\.((t|j)sx?)$",
    "verbose": true,
    "coverageReporters": [
      "lcov",
      "text",
      "json-summary"
    ],
    "reporters": [
      "default",
      [
        "jest-sonar",
        {
          "outputDirectory": ".",
          "outputName": "test-report.xml"
        }
      ]
    ],
    "testEnvironment": "jsdom"
  },
  "peerDependencies": {
    "prop-types": ">=15.0.0",
    "react": "^16.8.3 || ^17.0.0",
    "styled-components": "^5.1.0"
  },
  "devDependencies": {
    "@babel/core": "^7.14.6",
    "@babel/plugin-proposal-private-property-in-object": "^7.14.5",
    "@repay/babel-preset": "^1.1.1",
    "@repay/cactus-theme": "^2.1.0",
    "@repay/scripts": "^3.0.0",
    "@storybook/addon-knobs": "^6.3.0",
    "@storybook/react": "^6.3.2",
    "@storybook/theming": "^6.3.2",
    "@svgr/core": "^5.4.0",
    "@testing-library/react": "^12.0.0",
    "@types/jest": "^26.0.23",
    "@types/node": "^14.17.3",
    "@types/storybook__react": "^5.2.1",
    "@types/styled-components": "5.1.9",
    "@types/styled-system": "^5.1.11",
    "babel-jest": "^27.0.2",
    "babel-loader": "^8.2.2",
    "fast-glob": "^3.2.4",
    "jest": "^27.0.4",
    "jest-sonar": "^0.2.12",
    "jest-styled-components": "^7.0.4",
    "prop-types": "^15.7.2",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
    "styled-components": "^5.3.0",
    "typescript": "~4.2.4"
  },
  "dependencies": {
    "styled-system": "^5.1.5"
  }
}
