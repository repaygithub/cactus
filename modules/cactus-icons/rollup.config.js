module.exports = (config) => {
  config.plugins.splice(3, 1)
  return config
}
