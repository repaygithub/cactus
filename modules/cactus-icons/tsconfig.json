{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "i",
    "lib": ["esnext", "dom"],
    "types": []
  },
  "include": ["ts"]
}
