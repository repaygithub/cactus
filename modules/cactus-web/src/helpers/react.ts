import { assign, noop } from 'lodash'
import React, { SetStateAction } from 'react'

export type CloneFunc = (
  e: React.ReactElement,
  p?: Record<string, any>,
  ix?: number
) => React.ReactElement

// This can also be used just to flatten the children, but cloning seems the more likely use case.
export function cloneAll(
  children: React.ReactNode,
  cloneProps?: Record<string, any>,
  makeClone: CloneFunc = React.cloneElement
): React.ReactChild[] {
  const hasKeyPrefix = cloneProps && (cloneProps.key || cloneProps.key === 0)
  const childArray = React.Children.toArray(children) as React.ReactChild[]
  return childArray.reduce((returnVal, child, index) => {
    if (!child || typeof child === 'string' || typeof child === 'number') {
      returnVal.push(child)
    } else {
      const key = hasKeyPrefix ? `${cloneProps?.key}${child.key}` : child.key
      let props: Record<string, any> | undefined
      if (cloneProps) {
        props = { ...cloneProps, key }
      } else if (hasKeyPrefix) {
        props = { key }
      }
      if (child.type === React.Fragment) {
        if (child.props.children) {
          returnVal.push(...cloneAll(child.props.children, props, makeClone))
        }
      } else {
        returnVal.push(makeClone(child, props, index))
      }
    }
    return returnVal
  }, [] as React.ReactChild[])
}

type Ref<T> = React.RefCallback<T> | React.MutableRefObject<T> | null | undefined
type HybridRef<T> = React.RefCallback<T> & React.RefObject<T>
type MutHybridRef<T> = React.RefCallback<T> & React.MutableRefObject<T>

// Keep in mind that unlike a normal ref, hybrid refs are not guaranteed to be stable.
export function useMergedRefs<T>(...refs: Ref<T>[]): HybridRef<T> {
  const refFunc: any = (value: T | null) => {
    if (!value) return
    hybridRef.current = value
    for (const ref of refs) {
      if (!ref) {
        continue
      } else if (typeof ref === 'function') {
        ref(value)
      } else {
        ref.current = value
      }
    }
  }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  const hybridRef = React.useCallback<MutHybridRef<T>>(refFunc, refs)
  if (hybridRef === refFunc) {
    Object.defineProperty(hybridRef, 'current', { writable: true, enumerable: true })
    Object.preventExtensions(hybridRef)
  }
  return hybridRef
}

type Func<R = any> = (...args: any[]) => R
type NotFunc<T> = T extends Func ? never : T

interface ValueHook {
  <T extends Func>(mkValue: T, args: Parameters<T>): ReturnType<T>
  <T extends Func, D = Parameters<T>>(
    mkValue: T,
    deps: D,
    compareDependencies: (prev: Parameters<T>, current: D) => Parameters<T>
  ): ReturnType<T>
  <T>(value: NotFunc<T>, deps: unknown[]): T
  <T, D>(value: NotFunc<T>, deps: D, compareDependencies: (prev: D, current: D) => D): T
}

interface Value<T> {
  value: T
  dependencies: unknown[]
}

const shallowCmp = (prev: any[], current: any[]): any[] => {
  if (prev.length !== current.length || !prev.every((d, i) => d === current[i])) {
    return current
  }
  return prev
}

// Similar to useMemo/useCallback except stability is guaranteed.
export const useValue: ValueHook = <T>(val: any, deps: any[], cmp = shallowCmp): T => {
  const ref = React.useRef(SENTINEL as Value<T>)
  let value = ref.current.value
  const prevDeps = ref.current.dependencies
  const dependencies = cmp(prevDeps, deps)
  if (dependencies !== prevDeps) {
    value = typeof val === 'function' ? val(...dependencies) : val
    ref.current = { value, dependencies }
  }
  return value
}
const SENTINEL: unknown = { dependencies: [Object.create(null)] }

interface UseBox {
  <T>(box: T): T
  // Mutable box + immutable initialization
  <M, I extends Func>(box: M, init: I, ...args: Parameters<I>): M & ReturnType<I>
}
type Obj = Record<string, any>

// An immutable container with mutable contents; basically, a self-updating ref.
export const useBox: UseBox = (box: Obj, init?: Func<Obj>, ...args: any[]): Obj => {
  const ref = React.useRef<Obj>()
  if (ref.current === undefined) {
    ref.current = init ? init(...args) : {}
  }
  return assign(ref.current, box)
}

type StateSetterWithCallback<S> = (v: SetStateAction<S>, callback?: () => void) => void

export function useStateWithCallback<S>(
  initialState: S | (() => S)
): [S, StateSetterWithCallback<S>] {
  const [state, setState] = React.useState<S>(initialState)
  const storedCallback = React.useRef<() => void>(noop)

  const setStateWithCallback = React.useCallback<StateSetterWithCallback<S>>(
    (v, callback = noop) => {
      storedCallback.current = callback
      setState(v)
    },
    []
  )

  React.useEffect(() => {
    storedCallback.current()
    storedCallback.current = noop
  }, [state])

  return [state, setStateWithCallback]
}

const inc = (x: number) => x + 1

export const useRenderTrigger = (): (() => void) => React.useReducer(inc, 0)[1]

type SemiControlProps<T extends React.ElementType> = {
  input: T
  value: unknown
  onChange: React.ChangeEventHandler<unknown>
} & Omit<React.ComponentProps<T>, 'as' | 'ref' | 'value' | 'onChange'>

interface ControlBox {
  value: unknown
  isFocused: boolean
  onFocus?: React.FocusEventHandler<unknown>
  handleFocus: React.FocusEventHandler<unknown>
  onBlur?: React.FocusEventHandler<unknown>
  handleBlur: React.FocusEventHandler<unknown>
  onChange: React.ChangeEventHandler<unknown>
  handleChange: React.ChangeEventHandler<unknown>
}

const initControlBox = (trigger: () => void): ControlBox => {
  const vars: ControlBox = {
    value: null,
    isFocused: false,
    onChange: noop,
    handleFocus: (e: React.FocusEvent<unknown>) => {
      vars.isFocused = true
      vars.onFocus?.(e)
    },
    handleBlur: (e: React.FocusEvent<unknown>) => {
      vars.isFocused = false
      vars.onBlur?.(e)
      trigger() // Now rerender with the parent's value.
    },
    handleChange: (e: React.ChangeEvent<{ value: unknown }>) => {
      vars.value = e.target.value
      vars.onChange(e)
      trigger()
    },
  }
  return vars
}

// Use this to make an input "uncontrolled" while focused, but controlled otherwise;
// this is useful when valid values are limited but you don't want to interfere
// with user input by trying to normalize the value as they type.
export const SemiControlled = <T extends React.ElementType>(
  props: SemiControlProps<T>
): React.ReactElement => {
  const { input, value: controlledValue, onFocus, onBlur, onChange, ...rest } = props
  const trigger = useRenderTrigger()
  const vars = useBox({ onFocus, onBlur, onChange, controlledValue }, initControlBox, trigger)
  if (!vars.isFocused) {
    vars.value = controlledValue
  }
  return React.createElement(input, {
    ...rest,
    onFocus: vars.handleFocus,
    onBlur: vars.handleBlur,
    onChange: vars.handleChange,
    value: vars.value,
  })
}
