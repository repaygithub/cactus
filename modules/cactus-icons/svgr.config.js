const prettierConfig = JSON.parse(
  require('fs').readFileSync(require('path').join(__dirname, '../..', '.prettierrc'))
)
prettierConfig.parser = 'typescript'

const exportedTemplate = ({ template }, opts, { componentName, jsx }) => {
  const code = `
import PropTypes from 'prop-types'
import * as React from 'react'
import styled from 'styled-components'
import { color, compose, space, verticalAlign } from 'styled-system'
import iconSizes from './iconSizes'
import { IconProps } from './types'

const Base = ({ iconSize, verticalAlign, opacity, color, ...props }: IconProps) => {
  return (
    JSX
  )
}

const COMPONENT_NAME = styled(Base)(
  compose(
    space,
    color,
    verticalAlign,
    iconSizes
  )
)

COMPONENT_NAME.propTypes = {
  // @ts-ignore
  iconSize: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.array]),
}

COMPONENT_NAME.defaultProps = {
  verticalAlign: 'middle',
}

export default COMPONENT_NAME
  `
  const typeScriptTpl = template.smart(code, {
    plugins: ['typescript'],
    preserveComments: true,
    placeholderPattern: /^[_A-Z]{2,}$/,
  })

  return typeScriptTpl({ JSX: jsx, COMPONENT_NAME: componentName })
}

module.exports = {
  icon: true,
  ext: 'tsx',
  template: exportedTemplate,
  svgProps: {
    fill: 'currentcolor',
  },
  prettierConfig,
}
