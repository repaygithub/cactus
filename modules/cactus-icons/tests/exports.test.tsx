import * as icons from '@repay/cactus-icons'
import { cleanup } from '@testing-library/react'

afterEach(cleanup)

describe('Exports', () => {
  test('should match snapshot of exported icons', () => {
    expect(Object.keys(icons)).toMatchSnapshot()
  })
})
