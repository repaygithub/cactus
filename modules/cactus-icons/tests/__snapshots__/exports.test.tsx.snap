// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exports should match snapshot of exported icons 1`] = `
Array [
  "ActionsAdd",
  "ActionsCoachmarks",
  "ActionsCopy",
  "ActionsDelete",
  "ActionsDownload",
  "ActionsEdit",
  "ActionsFilter",
  "ActionsGear",
  "ActionsKey",
  "ActionsPrint",
  "ActionsRedo",
  "ActionsRefresh",
  "ActionsRemove",
  "ActionsUndo",
  "ActionsUpload",
  "BatchstatusInprogress",
  "BatchstatusOpen",
  "BatchstatusReconciled",
  "BatchstatusSettle",
  "DescriptiveAssistant",
  "DescriptiveAt",
  "DescriptiveBars",
  "DescriptiveBox",
  "DescriptiveCalendar",
  "DescriptiveCart",
  "DescriptiveClock",
  "DescriptiveCloud",
  "DescriptiveComment",
  "DescriptiveConnect",
  "DescriptiveDelivered",
  "DescriptiveDiscussions",
  "DescriptiveDocument1",
  "DescriptiveDocument2",
  "DescriptiveEnvelope",
  "DescriptiveFilter",
  "DescriptiveFolder",
  "DescriptiveHome",
  "DescriptiveInternet",
  "DescriptiveLocation",
  "DescriptiveLock",
  "DescriptivePalette",
  "DescriptivePiechart",
  "DescriptivePresentation",
  "DescriptiveProfile",
  "DescriptiveReceipt",
  "DescriptiveRuler",
  "DescriptiveSuitcase",
  "DescriptiveTray",
  "InterfaceCard",
  "InterfaceDesktop",
  "InterfaceLaptop",
  "InterfacePage",
  "InterfaceSms",
  "InterfaceTables",
  "MultimediaCall",
  "MultimediaCamera",
  "MultimediaVideocamera",
  "MultimediaVoice",
  "MultimediaVolume",
  "NavigationArrowDown",
  "NavigationArrowLeft",
  "NavigationArrowRight",
  "NavigationArrowUp",
  "NavigationChevronDown",
  "NavigationChevronLeft",
  "NavigationChevronNav",
  "NavigationChevronRight",
  "NavigationChevronUp",
  "NavigationCircleDown",
  "NavigationCircleLeft",
  "NavigationCircleRight",
  "NavigationCircleUp",
  "NavigationClose",
  "NavigationDragDrop",
  "NavigationFirst",
  "NavigationHamburger",
  "NavigationLast",
  "NavigationLess",
  "NavigationLogout",
  "NavigationMenuDots",
  "NavigationPlus",
  "NavigationSearch",
  "NotificationAlert",
  "NotificationError",
  "NotificationInfo",
  "NotificationNote",
  "PaymentAmex",
  "PaymentBank",
  "PaymentBarcode",
  "PaymentBatch",
  "PaymentCardBack",
  "PaymentCash",
  "PaymentCashback",
  "PaymentCharge",
  "PaymentCredit",
  "PaymentCsc",
  "PaymentCvv",
  "PaymentDebit",
  "PaymentInvoice",
  "PaymentMaster",
  "PaymentPay",
  "PaymentPaybycard",
  "PaymentPayment",
  "PaymentRefund",
  "PaymentToken",
  "PaymentWallet",
  "StatusApprove",
  "StatusCheck",
  "StatusDanger",
  "StatusFollow",
  "StatusLike",
  "StatusMoney",
  "StatusRecomend",
  "StatusSpinner",
  "StatusStar",
  "StatusStarFilled",
  "StatusSun",
  "StatusTarget",
  "StatusThumbsdown",
  "StatusThumbsup",
  "StatusTrophy",
  "StatusUnfollow",
  "iconSizes",
]
`;
